### PR TITLE
Add backend functionality to upload pictures

### LINKED ISSUE #4: Create backend upload functionality
US:
10 - I want to display pictures which were uploaded to my channel so that users can see their pictures.
(12 - I want to provide possibilities to upload new images to my channel so that I can display them.)

### PR DESCRIPTION
**Pull request by NoFrust**
Add a route /api/picture to the api which allows to upload pictures.

Resolves #4

### CODE REVIEW

File: api/api.js [hunk 1]
```diff
@@ -11,28 +12,57 @@ const apiVersion = 1;
 // App
 const app = express();
 
-app.get('/api/' + apiVersion + '/test', function(req, res) {
-  console.log('REQUEST ~~~ [GET] '.yellow  + 'GET TEST');
-  res.send('Hello world');
+//Handle form data file upload. Files accessible via req.files
+app.use(fileUpload());
+
+var pictures = require('./routes/picture');
```
Comment (tobuto) [inline api/api.js]: please use const or let instead of var (const in this context)

File: api/routes/picture.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+var fs = require('fs');
+var path = require('path');
+
+var express = require('express');
+var router = express.Router();
+
+var Picture = require('./../models/Picture');
+
+//get a specific picture
+router.get('/', function (req, res, next) {
```
Comment (tobuto) [inline api/routes/picture.js]: Maybe use a wildcard path instead of a query parameter? ('/:id')

File: api/routes/picture.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        res.status(200);
+        res.contentType(picture.img.contentType);
+        res.send(picture.img.data);
+    });
+});
+
+//upload a picture
+router.post('/', function (req, res, next) {
+    if (req.body.screen == undefined) {
+        var error = new Error('Please specify a valid picture id.');
```
Comment (tobuto) [inline api/routes/picture.js]: 'Please specify a valid screen id.'

File: api/routes/picture.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        return next(error);
+    }
+
+    if (req.files.image == undefined) {
+        var error = new Error('Please provide an image.');
+        error.status = 409;
+        return next(error);
+    }
+
+    var pic = new Picture;
```
Comment (tobuto) [inline api/routes/picture.js]: const

File: api/routes/picture.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    pic.save(function (err, image) {
+        if (err) return next(err);
+        res.status(200);
+        res.json({id: image._id, screen: image.screen});
+    });
+
+});
+
+//Delete a specific picture
+router.delete('/', function (req, res, next) {
```
Comment (tobuto) [inline api/routes/picture.js]: '/:id' ?

File: api/routes/picture.js (hunk 2 above)
Comment (NoFrust) [inline api/routes/picture.js]: This change would be a conflict with the method to get all pictures of a certain screen. My proposal is to use wildcard paths for /api/pictures and use a new route /api/screen to provide the screen functionality. What do you think?

Comment (tobuto) [inline api/routes/picture.js]: Sounds good

**Comment by NoFrust** [comment]
Requested changes accepted and implemented

**Pull request merged**